<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from mi_br.urdf.xacro               | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="simple_arm_3dof">
  <!-- ============================= -->
  <!--   LINK 0: Base                -->
  <!-- ============================= -->
  <link name="base_link">
    <visual>
      <geometry>
        <cylinder length="0.1" radius="0.05"/>
      </geometry>
      <material name="gray">
        <color rgba="0.5 0.5 0.5 1"/>
      </material>
    </visual>
  </link>
  <!-- ============================= -->
  <!--   LINK 1: Primer brazo         -->
  <!-- ============================= -->
  <link name="link1">
    <visual>
      <!-- Centrado para que la unión quede en el borde -->
      <origin xyz="0 0 0.15"/>
      <geometry>
        <box size="0.05 0.05 0.3"/>
      </geometry>
      <material name="blue">
        <color rgba="0 0 1 1"/>
      </material>
    </visual>
  </link>
  <joint name="joint1" type="revolute">
    <parent link="base_link"/>
    <child link="link1"/>
    <origin xyz="0 0 0.05"/>
    <axis xyz="0 0 1"/>
    <!-- Rotación en plano horizontal -->
    <limit effort="1.0" lower="-1.57" upper="1.57" velocity="1.0"/>
  </joint>
  <!-- ============================= -->
  <!--   LINK 2: Segundo brazo        -->
  <!-- ============================= -->
  <link name="link2">
    <visual>
      <origin xyz="0 0 0.1"/>
      <geometry>
        <box size="0.05 0.05 0.2"/>
      </geometry>
      <material name="green">
        <color rgba="0 1 0 1"/>
      </material>
    </visual>
  </link>
  <joint name="joint2" type="revolute">
    <parent link="link1"/>
    <child link="link2"/>
    <origin xyz="0 0 0.3"/>
    <!-- Final de link1 -->
    <axis xyz="0 1 0"/>
    <!-- Rotación vertical tipo codo -->
    <limit effort="1.0" lower="-1.57" upper="1.57" velocity="1.0"/>
  </joint>
  <!-- ============================= -->
  <!--   LINK 3: Tercer brazo (muñeca)-->
  <!-- ============================= -->
  <link name="link3">
    <visual>
      <origin xyz="0 0 0.1"/>
      <geometry>
        <box size="0.04 0.04 0.2"/>
      </geometry>
      <material name="red">
        <color rgba="1 0 0 1"/>
      </material>
    </visual>
  </link>
  <joint name="joint3" type="revolute">
    <parent link="link2"/>
    <child link="link3"/>
    <origin xyz="0 0 0.2"/>
    <!-- Final de link2 -->
    <axis xyz="0 1 0"/>
    <!-- Otro pitch para la muñeca -->
    <limit effort="1.0" lower="-1.57" upper="1.57" velocity="1.0"/>
  </joint>


</robot>
